### PR TITLE
Bind sql parameters spreadsheet (#2095)

### PR DESCRIPTION
**Pull request by kristophesankar**
PR for issue #2095 for spreadsheet.inc.php

### CODE REVIEW

File: library/spreadsheet.inc.php [hunk 1]
```diff
@@ -124,31 +124,46 @@ function real2form($fldval)
...
-            sqlStatement("UPDATE form_$spreadsheet_form_name SET "      .
-            "value = '$form_completed|$start_date|$template_name' " .
-            "WHERE id = '$formid' AND rownbr = -1 AND colnbr = -1");
-            sqlStatement("DELETE FROM form_$spreadsheet_form_name WHERE " .
-              "id = '$formid' AND rownbr >= 0 AND colnbr >= 0");
+            sqlStatement(
+                "UPDATE " . escape_table_name('form_' . $spreadsheet_form_name) .
+                " SET value = ? WHERE id = ? AND rownbr = -1 AND colnbr = -1",
+                array(
+                    $form_completed|$start_date|$template_name,
```
Comment (bradymiller) [inline library/spreadsheet.inc.php]: Change to:
```
$form_completed . '|' . $start_date . '|' . $template_name
```

File: library/spreadsheet.inc.php [hunk 2]
```diff
@@ -124,31 +124,46 @@ function real2form($fldval)
...
-              "$formid, -1, -1, 0, " .
-              "'$form_completed|$start_date|$template_name' " .
-              ")");
+            sqlInsert(
+                "INSERT INTO " . escape_table_name('form_' .$spreadsheet_form_name) . " ( " .
+                "id, rownbr, colnbr, datatype, value " .
+                ") VALUES ( ?, -1, -1, 0, ? )",
+                array(
+                    $formid,
+                    $form_completed|$start_date|$template_name
```
Comment (bradymiller) [inline library/spreadsheet.inc.php]: change to:
```
$form_completed . '|' . $start_date . '|' . $template_name
```

File: library/spreadsheet.inc.php [hunk 3]
```diff
@@ -166,9 +181,11 @@ function real2form($fldval)
         // which must not match any existing template name.
         $new_template_name = form2real($_POST['form_new_template_name']);
         if ($new_template_name != $template_name) {
-            $trow = sqlQuery("SELECT id FROM form_$spreadsheet_form_name WHERE " .
-            "id < 0 AND rownbr = -1 AND colnbr = -1 AND value = '" .
-            real2db($new_template_name) . "'");
+            $trow = sqlQuery(
+                "SELECT id FROM " . escape_table_name('form_' . $spreadsheet_form_name) .
+                " WHERE id < 0 AND rownbr = -1 AND colnbr = -1 AND value = ?",
+                array(real2db($new_template_name))
```
Comment (bradymiller) [inline library/spreadsheet.inc.php]: Remove the call to real2db(), this looks like a legacy way to try to escape for database. We can also remove the call below and remove this real2db() function entirely near the top of this script (demolition is always fun :) ).

File: library/spreadsheet.inc.php [hunk 4]
```diff
@@ -181,23 +198,33 @@ function real2form($fldval)
...
-                "$tempid, -1, -1, 0, " .
-                "'" . real2db($template_name) . "' " .
-                ")");
+                sqlInsert(
+                    "INSERT INTO " . escape_table_name('form_' . $spreadsheet_form_name) . " ( " .
+                    "id, rownbr, colnbr, datatype, value " .
+                    ") VALUES ( ?, -1, -1, 0, ? )",
+                    array(
+                        $tempid,
+                        real2db($template_name)
```
Comment (bradymiller) [inline library/spreadsheet.inc.php]: remove the call to this real2db function

**Comment by bradymiller** [comment]
Nice work @kristophesankar . Just left comments with several minor things to address. After you address those, then will be ready for codebase. thanks, -brady

File: library/spreadsheet.inc.php (hunk 1 above)
Comment (kristophesankar) [reply library/spreadsheet.inc.php]: I thought it was a bitwise operation initially. This makes more sense though. Thanks.

File: library/spreadsheet.inc.php (hunk 3 above)
Comment (kristophesankar) [reply library/spreadsheet.inc.php]: I'll remove them. There doesn't seem to be any more references to it from this file. Its also just a container function for `addslashes` so we should be safe to remove it.

**Comment by bradymiller** [comment]
hi @kristophesankar , Code looks good and bringing into codebase. Thanks for the contribution! -brady

**Comment by kristophesankar** [comment]
Great @bradymiller ! If there's any more work to do I'll be happy to lend a hand.

**Comment by bradymiller** [comment]
hi @kristophesankar ,

We always have work to do :)

That GeoIP filtering feature does look really neat: https://github.com/openemr/openemr/issues/2098

Check that out and see if your interested. Otherwise, I can definitely find other things that can be worked on.

thanks, -brady

**Comment by kristophesankar** [comment]
Thanks :) I'll definitely checkout what's on the issues page and let you all know what I'd like to work on. This is a cool project that I've been familiar with for a few years, so for me, its great to help out.

**Pull request merged**